optimizely/bucketer.py

```diff
@@ -78,12 +90,15 @@ def find_bucket(self, project_config, bucketing_id, parent_id, traffic_allocatio
 
         for traffic_allocation in traffic_allocations:
             current_end_of_range = traffic_allocation.get('endOfRange')
-            if bucketing_number < current_end_of_range:
+            if current_end_of_range is not None and bucketing_number < current_end_of_range:
                 return traffic_allocation.get('entityId')
 
         return None
 
-    def bucket(self, project_config, experiment, user_id, bucketing_id):
+    def bucket(
+        self, project_config: ProjectConfig,
+        experiment: Experiment, user_id: str, bucketing_id: str
+    ) -> tuple[None | dict[None, None] | Variation, list[str]]:
```

Comment:
Is this correct? `dict[None, None]`

Reply:
So `get_variation_from_id_by_experiment_id` and `get_variation_from_key_by_experiment_id` use an empty dict as their null case, which is confusing because a `Variation` is not a dict. I used this type signature to make that more explicit.

I think they should ultimately be changed to return `None` instead.

Reply:
Let's simply it and use tuple with Optional Variation and list of str.